boards/same54-xpro/board.c

```diff
@@ -97,3 +97,13 @@ VFS_AUTO_MOUNT(lwext4, VFS_MTD(sdhc_dev), VFS_DEFAULT_SD(0), 1);
 #endif
 #endif /* MODULE_VFS_DEFAULT */
 #endif /* MODULE_SAM0_SDHC */
+
+#ifdef MODULE_PERIPH_FREQM
+#include "periph/freqm.h"
+    struct freqm_config config_freqm = {
+        .msr_clk_src = SAM0_GCLK_MAIN,
+        .ref_clk_src = SAM0_GCLK_32KHZ,
+        .ref_clk_cnt = 128,
+        .ref_clk_freq = 32768
```

Comment:
> Are the pins used by the frequency meter fixed? How do we select the input?

The documentations states that the reference clock and the measurement clock can be selected from the available GCLK_FREQM_REF respectively GCLK_FREQM_MSR sources. Further the GCLK I/O lines (GCLK_IO[7:0]) can be used as measurement or reference clock sources.
So the Peripheral Channels (PCHCTRLm) 5 (FREQM Measure) and 6 (FREQM Reference) have to configured for the input. This makes the GCLK I/O lines (GCLK_IO[7:0]) also available as sources for reference or measuremnt. I didn't implement using pins as I thought this is not part of the drive, because the user can configure the clock generator and select it: 

Input is selected by selecting one of the 12 generators by writing to `GCLK->PCHCTRL[5].reg` for the measurement clock and `GCLK->PCHCTRL[6].reg` for the reference clock.

Reply:
While measuring internal clocks might surely be interesting at times, I would imagine the main use case of this driver / interface to be about measuring external signals.

While yes, a user could manually set the MUX of an eligible GPIO to `MUX_M` for GCLK and configure the associated GCLK as input themself, that's neither hardware independent nor trivial and the frequency meter driver should take care of it instead.